examples/ensemble/plot_voting_decision_regions.py

```diff
@@ -5,69 +5,172 @@
+)
+clf3 = make_pipeline(
+    StandardScaler(),
+    Nystroem(gamma=2, random_state=0),
+    LogisticRegression(C=10),
+)
+weights = [2, 1, 3]
 eclf = VotingClassifier(
-    estimators=[("dt", clf1), ("knn", clf2), ("svc", clf3)],
+    estimators=[("constant", clf1), ("periodic", clf2), ("nystroem", clf3)],
     voting="soft",
-    weights=[2, 1, 2],
+    weights=weights,
 )
 
 clf1.fit(X, y)
 clf2.fit(X, y)
 clf3.fit(X, y)
 eclf.fit(X, y)
 
-# Plotting decision regions
-f, axarr = plt.subplots(2, 2, sharex="col", sharey="row", figsize=(10, 8))
-for idx, clf, tt in zip(
+# %%
+# Finally we use :class:`~inspection.DecisionBoundaryDisplay` to plot the
+# predicted probabilities. By using a diverging colormap (such as `"RdBu"`), we
+# can ensure that darker colors correspond to `predict_proba` close to either 0
+# or 1.
+
+from itertools import product
+
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap
+
+from sklearn.inspection import DecisionBoundaryDisplay
+
+fig, axarr = plt.subplots(2, 2, sharex="col", sharey="row", figsize=(10, 8))
+for idx, clf, title in zip(
     product([0, 1], [0, 1]),
     [clf1, clf2, clf3, eclf],
-    ["Decision Tree (depth=4)", "KNN (k=7)", "Kernel SVM", "Soft Voting"],
+    [
+        "Splines with\nconstant extrapolation",
+        "Splines with\nperiodic extrapolation",
+        "RBF Nystroem",
+        "Soft Voting",
+    ],
 ):
-    DecisionBoundaryDisplay.from_estimator(
-        clf, X, alpha=0.4, ax=axarr[idx[0], idx[1]], response_method="predict"
+    disp = DecisionBoundaryDisplay.from_estimator(
+        clf,
+        X,
+        response_method="predict_proba",
+        plot_method="pcolormesh",
+        cmap="RdBu",
+        alpha=0.8,
+        ax=axarr[idx[0], idx[1]],
     )
-    axarr[idx[0], idx[1]].scatter(X[:, 0], X[:, 1], c=y, s=20, edgecolor="k")
-    axarr[idx[0], idx[1]].set_title(tt)
+    axarr[idx[0], idx[1]].scatter(
+        X["Feature #0"],
+        X["Feature #1"],
+        c=y,
+        cmap=ListedColormap(["tab:red", "tab:blue"]),
+        edgecolor="white",
+        linewidth=1,
+    )
+    axarr[idx[0], idx[1]].set_title(title)
+    fig.colorbar(disp.surface_, ax=axarr[idx[0], idx[1]], label="Probability estimate")
 
 plt.show()
+
+# %%
+# As a sanity check, we can verify for a given sample that the probability
+# predicted by the :class:`~ensemble.VotingClassifier` is indeed the weighted
+# average of the individual classifiers' soft-predictions.
+#
+# In the case of binary classification such as in the present example, the
+# `predict_proba` arrays contain the probability of belonging to class 0 (here
+# in red) as the first entry, and the probability of belonging to class 1 (here
+# in blue) as the second entry.
+
+test_sample = pd.DataFrame({"Feature #0": [-0.5], "Feature #1": [1.5]})
+predict_probas = [est.predict_proba(test_sample).ravel() for est in eclf.estimators_]
+print(f"Individual predicted probabilities: {predict_probas}")
```

Comment:
I think the output would be easier to read by using several print statements in a loop:

```suggestion
predict_probas = [est.predict_proba(test_sample).ravel() for est in eclf.estimators_]
for (est_name, _), est_probas in zip(eclf.estimators, predict_probas):
    print(f"{est_name}'s predicted probabilities: {est_probas}")
```